### PR TITLE
Service related changes

### PR DESCRIPTION
**Pull request by arunkumars08**
1. missions integrates with prod-preview 
2. runtimes integrates with prod-preview
3. pipelines integrates with prod-preview
4. installs ngx-login-client for auth service
5. handles generic service to fetch auth, url, and origin (From OSIO or Launcher)
6. Mocks and injects values to run standalone

### CODE REVIEW

**Comment by edewit** [comment]
Cool, now all we need is some tests, right?

**Comment by invincibleJai** [comment]
@edewit we are working on it , to start with fixing existing broken test.

**Comment by edewit** [comment]
cool

File: src/app/launcher/service/common.service.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    getOrigin(): string {
+        if (this.config) {
+            return this.config.get(this.keys.ORIGIN);
+        }
+        return null;
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (edewit) [reply src/app/launcher/service/common.service.ts]: all we need to pass is the Barear Token in the current code we have this [TokenProvider](https://github.com/fabric8-launcher/ngx-launcher/blob/master/src/app/service/token-provider.ts)

File: src/app/launcher/release-strategy-step/release-strategy-step.component.ts [hunk 2]
```diff
@@ -78,7 +78,13 @@ export class ReleaseStrategyStepComponent extends WizardStep implements OnInit,
 
     this.wizardComponent.addStep(this);
     this.subscriptions.push(this.pipelineService.getPipelines().subscribe((result) => {
-      this._pipelines = this.allPipelines = result;
+      // needs to filter out associated pipelines from list of pipelines
+      let selPipelines: any[] = [];
+      selPipelines = result.filter(item => {
+        return item.platform == "maven";
+      })
+
```
[the remark this answers is not in the document]
Comment (invincibleJai) [reply src/app/launcher/release-strategy-step/release-strategy-step.component.ts]: @dlabrecq Dan, it should be filtered based on platform i.e maven, node, pypi but in current endpoints we are not getting "platform" in runtimes api [/api/booster-catalog/runtimes], but for pipelines endpoint we get pipelines for all platform , that is the reason having "maven" for now will update Runtime model once  "platform"  is available and default value is "maven" as per discussion here https://groups.google.com/forum/#!topic/fabric8-launcher/VIysVm_-tGc

**Comment by arunkumars08** [comment]
@dlabrecq I have still kept the helper.service.ts (earlier common.service.ts) under launcher services, as it just has some generic functions that can be used from both launcher and osio flow. Please let me know if you think otherwise. And regarding AuthenticationService, whatever you said made sense hence I removed the ngx-login-client include and used TokenProvider as that already has generic implementation of getting tokens.

**Comment by dlabrecq** [comment]
Looks good. 

The only thing that seems to be missing is an export from index.ts? The class won't be available to the provider implementation in OSIO. Although, that fix can be another PR.

**Pull request merged**